File: Benchmarks/Core/Components/TileMapComponent2D.cs
using Benchmarks.Core.Collections;
using Benchmarks.Core.Types;

namespace Benchmarks.Core.Components
{
    public struct TileLayer
    {
        public string Name { get; set; }
        public int[] Data { get; set; }
    }

    public struct TileMapComponent2D
    {
        public int Width { get; set; }
        public int Height { get; set; }
        public int TileWidth { get; set; }
        public int TileHeight { get; set; }
        public YTB<TileLayer> TileLayers { get; set; }
        public Dictionary<int, List<Rectangle>> Collisions { get; set; }
    }
}

File: Benchmarks_FSharp/Core/Components/TileMapComponent2D.fs
namespace Benchmarks.Core.Components

open System.Collections.Generic
open Benchmarks.Core.Collections
open Benchmarks.Core.Types

[<Struct>]
type TileLayer =
    val mutable Name: string
    val mutable Data: int array

[<Struct>]
type TileMapComponent2D =
    val mutable Width: int
    val mutable Height: int
    val mutable TileWidth: int
    val mutable TileHeight: int
    val mutable TileLayers: YTB<TileLayer>
    val mutable Collisions: Dictionary<int, List<Rectangle>>
